overlord/snapstate/autorefresh.go

```diff
@@ -128,6 +128,31 @@ func (m *autoRefresh) EffectiveRefreshHold() (time.Time, error) {
 	return holdTime, nil
 }
 
+func (m *autoRefresh) ensureRefreshHoldAtLeast(duration time.Duration) error {
+	now := time.Now()
+
+	// get the effective refresh hold and check if it is sooner than the
+	// specified duration in the future
+	effective, err := m.EffectiveRefreshHold()
+	if err != nil {
+		return err
+	}
+
+	if effective.IsZero() || effective.Sub(now) < duration {
+		// the effective refresh hold is sooner than the desired delay, so
+		// move it out to the specified duration
```

Comment:
not sure it's confusing or not, but if the effecitve was clumped because of max-postponement this is a no-op and should be